content/blogposts/2023/guest-post-arafat-khan-php-python.md

```diff
@@ -0,0 +1,256 @@
+
+
+Cody can be surprisingly helpful even when you just want to write all the code yourself. One of the classic mistakes while writing PHP as a newbie is missing the `$` sign before variables, as shown below. With Cody, it’s easy just to ask it to fix them for you, and you are good to go. It’s much better than the natural IntelliSense fixes that come with VS Code by default. I appreciate this feature as it saves me from tedious single-character bug fixes.
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/guest-post-arafat-khan-php-python-image-5-full.png"
+  caption="PHP Variable fixes with Cody"
+/>
+
+The other thing that I really enjoyed was the improved speed of coding.
+
+Typically, the coding process goes like this:
+
+1. Conceptualize
+2. Draft an imperfect code
+3. Refine
+4. Comment
+    
+However, with Cody, I experienced a much faster approach:
+
+1. Conceptualize
+2. Comment
+3. Generate a near-perfect code
+4. Refine
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/image_6.jpg"
+  caption="Couldn’t resist adding an AI image to show the contrast of two Coders"
+/>
+
+Sometimes, I would just describe what the function is “supposed” to do in a comment, and just like magic, Cody would write it for me.
+            <div>
+                <YouTube title="Autocomplete with Cody" id="UdMCERz9hcY" />
+            </div>
+The above example is a classic Cody’s autocomplete feature. It works surprisingly well for all kinds of code completion tasks, especially when your repo already has a reasonable amount of code to give it context. This autocomplete system is grounded in a complex foundation, and for those interested in its intricacies, you can delve deeper [here](https://about.sourcegraph.com/blog/the-lifecycle-of-a-code-ai-completion).
+
+All these features, such as _autocomplete, explanation, code fixes, and testing,_ made my job as a programmer a lot easier. I was able to build the building blocks of my code, and then, inch by inch, I was getting closer to completing the entire project. As I was building more and more of this codebase, a question that crossed my mind was:
+
+## Why use Cody when you have Chat GPT-4?
+
+I like the speed and convenience of Cody but I can’t quite tell why using the Vscode extension is so much better than using ChatGPT if I just wanted to ask code questions. This is where the real magic of Cody comes in. Cody can answer questions about your entire codebase and sift through all the different parts to find the most relevant information pertaining to your question. It uses the [Claude 100K model](https://www.anthropic.com/index/100k-context-windows) (soon to be the [200K model](https://www.anthropic.com/index/claude-2-1)) so it has the capacity to fit in so much more context than my copy-pasting hacks ever could. Take, for example, this question
```

Comment:
I like the speed and convenience of Cody but I can’t quite tell why using the Vscode extension is so much better than using ChatGPT if I just wanted to ask code questions. This is where the real magic of Cody comes in. =>

I like the speed and convenience of Cody, as well as the ability to use it directly in VS Code, but there is another major reason why I prefer Cody.